src/bus/webSocketConnection.spec.js

```diff
@@ -36,6 +39,335 @@ describe('WebScoketConnection', function() {
+            expectedOrganization.id
+          );
+
+          jsonRpcId = ws._jsonRpcId;
+
+          expectedJsonRpc = JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'MessageBus.Authorize',
+            params: {
+              token
+            },
+            id: jsonRpcId
+          });
+
+          promise = ws.authorize(token);
+
+          webSocketServer.on('connection', (socket) => {
+            socket.send(JSON.stringify(expectedMessage));
+          });
+        });
+
+        it('sends a message to the message bus', function() {
+          return promise.then(() => {
+            expect(send).to.be.calledWith(expectedJsonRpc);
+          });
+        });
+
+        it('increments the jsonRpcId', function() {
+          return promise.then(() => {
+            expect(ws._jsonRpcId).to.equal(jsonRpcId + 1);
+          });
+        });
+
+        it('fulfills the promise', function() {
+          expect(promise).to.be.fulfilled;
+        });
+
+        it('recieves a message from the server', function() {
+          return promise.then((response) => {
+            expect(response.authorized).to.be.true;
+            expect(response.error).to.be.null;
+          });
+        });
+      });
+
+      context('when a user is not authorized', function() {
+        let expectedMessage;
+        let expectedOrganization;
+        let expectedJsonRpc;
+        let jsonRpcId;
+        let promise;
+        let send;
+        let token;
+        let ws;
+
+        beforeEach(function() {
+          expectedMessage = {
+            jsonrpc: '2.0',
+            id: '0',
+            error: 'user is not authorized'
+          };
+          expectedOrganization = fixture.build('organization');
+          send = this.sandbox.stub(expectedWebSocket, 'send');
+          token = faker.random.uuid();
+
+          ws = new WebSocketConnection(
+            expectedWebSocket,
+            expectedOrganization.id
+          );
+
+          jsonRpcId = ws._jsonRpcId;
+
+          expectedJsonRpc = JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'MessageBus.Authorize',
+            params: {
+              token
+            },
+            id: jsonRpcId
+          });
+
+          promise = ws.authorize(token);
+
+          webSocketServer.on('connection', (socket) => {
+            socket.send(JSON.stringify(expectedMessage));
+          });
+        });
+
+        it('sends a message to the message bus', function() {
+          return promise.then(() => {
+            expect(send).to.be.calledWith(expectedJsonRpc);
+          });
+        });
+
+        it('increments the jsonRpcId', function() {
+          return promise.then(() => {
+            expect(ws._jsonRpcId).to.equal(jsonRpcId + 1);
+          });
+        });
+
+        it('fulfills the promise', function() {
+          expect(promise).to.be.fulfilled;
+        });
+
+        it('recieves a message from the server', function() {
```

Comment:
I'd only test the differences in this failure state, not the stuff that happens either way